coffea/nanoevents/mapping.py

```diff
@@ -21,7 +36,42 @@ def open_uuid(self, uuid):
         return rootdir
 
 
-class UprootSourceMapping(Mapping):
+# IMPORTANT -> For now the uuid is just the uuid of the pfn.
+#              Later we should use the ParquetFile common_metadata to populate.
+class TrivialParquetOpener(UUIDOpener):
+
+    class UprootLikeShim:
+        def __init__(self, file):
+            self.file = file
+
+        def read(self, column_name):
+            return self.file.read([column_name])
+
+        # for right now spoof the notion of directories in files
+        # parquet can do it but we've gotta convince people to
+        # use it like that
+        def __getitem__(self, name):
+            return self.file
+
+    def __init__(self, uuid_pfnmap, parquet_options={}):
+        super(TrivialParquetOpener, self).__init__(uuid_pfnmap)
+        self._parquet_options = parquet_options
+        self._schema_map = None
+
+    def open_uuid(self, uuid):
+        pfn = self._uuid_pfnmap[uuid]
+        parfile = pq.ParquetFile(pfn, **self._parquet_options)
+        pqmeta = parfile.schema_arrow.metadata
+        pquuid = None if pqmeta is None else pqmeta.get(b'uuid', None)
+        pfn = None if pqmeta is None else pqmeta.get(b'url', None)
+        if str(pquuid) != uuid:
+            raise RuntimeError(
+                f"UUID of file {pfn} does not match expected value ({uuid})"
+            )
+        return TrivialParquetOpener.UprootLikeShim(parfile)
+
+
+class BaseSourceMapping(Mapping):
```

Comment:
So I see you abstracted by subclassing, I was originally thinking about nesting: everything but `!load` would be in a "DSL interpreter" class and it would be constructed with the source map class (which only implements the `!load` via getitem) We can save that for another time though: my motivation was to implement a skim layer (basically the form part of the persistent key would start with a `!filter` expression, referencing another column that was joined with the existing source using a "split-source" mapping beneath the dsl interpreter layer)

Reply:
Seems reasonable and not hard - can do it in a later PR since this is future functionality.